src/mappings/disputeManager.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020-2022 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  DisputeFinalizedEvent,
+  DisputeOpenEvent,
+} from '@subql/contract-sdk/typechain/DisputeManager';
+import { FrontierEvmEvent } from '@subql/frontier-evm-processor';
+import assert from 'assert';
+import { Disputes, DisputeState, DisputeType } from '../types';
+
+function getDisputeType(type: number): DisputeType {
+  switch (type) {
+    case 0:
+      return DisputeType.POI;
+    case 1:
+      return DisputeType.QUERY;
+    default:
+      throw new Error(
+        `Unexpected dispute type "${type}" provided to function getDisputeType`
+      );
+  }
+}
+
+function getDisputeState(state: number): DisputeState {
+  switch (state) {
+    case 0:
+      return DisputeState.ONGOING;
+    case 1:
+      return DisputeState.ACCEPTED;
+    case 2:
+      return DisputeState.REJECTED;
+    case 3:
+      return DisputeState.CANCELLED;
+    default:
```

Comment:
[Suggest] Suggest to use `mapping` util instead of switch case. For example, In future, if there are `4` then we need add extra two lines of code to the switch codes.